dev/com.ibm.ws.logging/src/com/ibm/ws/logging/internal/impl/BaseTraceService.java

```diff
@@ -1205,6 +1205,159 @@ public TeePrintStream(TrOutputStream trStream, boolean autoFlush) {
             super(trStream, autoFlush);
             this.trStream = trStream;
         }
+
+        @Override
+        public synchronized void print(boolean b) {
+            TrOutputStream.isPrinting.set(true);
+            super.print(b);
+            TrOutputStream.isPrinting.set(false);
```

Comment:
should wrap super.print in a try block, with isPrinting.set(false) in a finally block, to ensure any exceptions thrown from super.print don't cause isPrinting.set(false) to be skipped.

this applies to all other similar print methods too, of course